applications/GeoMechanicsApplication/custom_utilities/dof_utilities.h

```diff
@@ -15,30 +15,70 @@
 #include <algorithm>
 #include <vector>
 
+#include "geo_aliases.h"
 #include "geometries/geometry.h"
 #include "includes/dof.h"
 #include "includes/node.h"
+#include "includes/variables.h"
 
 namespace Kratos::Geo::DofUtilities
 {
 
 std::vector<std::size_t> ExtractEquationIdsFrom(const std::vector<Dof<double>*>& rDofs);
 
+template <typename InputIt, typename OutputIt>
+OutputIt ExtractDofsFromNodes(InputIt                 NodeRangeBegin,
+                              InputIt                 NodeRangeEnd,
+                              OutputIt                DofPtrRangeBegin,
+                              const Variable<double>& rDofVariable)
+{
+    return std::transform(NodeRangeBegin, NodeRangeEnd, DofPtrRangeBegin, [&rDofVariable](const auto& r_node) {
+        return r_node.pGetDof(rDofVariable);
+    });
+}
+
 template <typename InputIt>
-std::vector<Dof<double>*> ExtractDofsFromNodes(InputIt                 NodePtrRangeBegin,
-                                               InputIt                 NodePtrRangeEnd,
+std::vector<Dof<double>*> ExtractDofsFromNodes(InputIt                 NodeRangeBegin,
+                                               InputIt                 NodeRangeEnd,
                                                const Variable<double>& rDofVariable)
 {
     auto result = std::vector<Dof<double>*>{};
-    std::transform(NodePtrRangeBegin, NodePtrRangeEnd, std::back_inserter(result),
-                   [&rDofVariable](const auto& r_node) { return r_node.pGetDof(rDofVariable); });
+    ExtractDofsFromNodes(NodeRangeBegin, NodeRangeEnd, std::back_inserter(result), rDofVariable);
+    return result;
+}
+
+template <typename NodeRange>
+std::vector<Dof<double>*> ExtractDofsFromNodes(const NodeRange& rNodes, const Variable<double>& rDofVariable)
+{
+    return ExtractDofsFromNodes(std::begin(rNodes), std::end(rNodes), rDofVariable);
+}
+
+template <typename NodeRange1, typename NodeRange2>
+std::vector<Dof<double>*> ExtractUPwDofsFromNodes(const NodeRange2& rSecondOrderNodes,
```

Comment:
Although I understand the logic of NodeRange2 for rSecondOrderNodes, it is confusing as these are the first argument and the U D.O.F. are the first D.O.F. in the element. NodeRangeDisplacementNodes and NodeRangeWaterPressureNodes may be clearer, but also lengthy.

Reply:
That is a fair point. I have renamed the template types to `DisplacementNodeRange` and `WaterPressureRange`. The function parameters I have renamed to `rDisplacementNodes` and `rWaterPressureNodes`. What do you think about that?